kong/llm/drivers/azure.lua

```diff
@@ -118,10 +118,18 @@ function _M.configure_request(conf)
     parsed_url = socket_url.parse(url)
   end
 
+  -- if the path is read from a URL capture, ensure that it is valid
+  parsed_url.path = string_gsub(parsed_url.path, "^/*", "/")
+
+  -- Azure-specific - prepend the deployment instance path to the
+  --                  user-override path.
+  if conf.model.options and conf.model.options.upstream_path then
+    conf.model.options.upstream_path = parsed_url.path
```

Comment:
The dynamic change conf in runtime is wired for me.  we need to rebuilt route in here I think
```
  if conf.model.options.upstream_url then
    parsed_url = socket_url.parse(conf.model.options.upstream_url)
  else
    -- azure has non-standard URL format
    local url = fmt(
      "%s%s",
      ai_shared.upstream_url_format[DRIVER_NAME]:format(conf.model.options.azure_instance, conf.model.options.azure_deployment_id),
          conf.model.options
      and conf.model.options.upstream_path
      or ai_shared.operation_map[DRIVER_NAME][conf.route_type]
      and ai_shared.operation_map[DRIVER_NAME][conf.route_type].path
      or "/"
    )
    parsed_url = socket_url.parse(url)
  end
```